### PR TITLE
Add Datadog client logs

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Add Datadog client logs

Log each message to console in dev env.

Add shop session ID to logger context.

## Justify why they are needed

This seems to be the official solution to log from the client.
https://docs.datadoghq.com/logs/log_collection/javascript/

Or is there another way to accomplish this?

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #484** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/484" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/484?utm_source=stack-comment).

File: apps/store/src/services/datadog.ts [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
     })
 
     datadogRum.startSessionReplayRecording()
+
+    datadogLogs.init({
+      ...COMMON_CONFIG,
+      clientToken: CLIENT_TOKEN,
+      forwardErrorsToLogs: true,
+
+      ...(ENV === 'dev'
```
Comment (alebedev) [inline apps/store/src/services/datadog.ts:35]: Do we want datadog logs when running tests?  Maybe `!== 'prod'` if we do

Comment (alebedev) [inline apps/store/src/services/datadog.ts:35]: I think multiline ternaries are not very readable and increase rate of programmer errors.  What about this way to define config?
```
const datadagConfig = {
  ...COMMON_CONFIG,
  // ...
}
if (ENV === 'dev') {
  datadogConfig.beforeSend = ...
}
datadogLogs.init(datadogConfig)
```

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/services/datadog.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/services/datadog.ts:35]: If you don't define env variables "APPLICATION_ID && CLIENT_TOKEN" then datadog is not initialized :)

Comment (robinandeer) [reply apps/store/src/services/datadog.ts:35]: Hmm kind of agree, I can update

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/484?utm_source=stack-comment).


Job Reference: N8bhvXinkGcsSWxBHKHr

**Pull request merged**